### PR TITLE
React Bookstore: Milestone 2-Components and Redux 

### PR DESCRIPTION
**Pull request by ferbaco86**
On this branch, we fulfilled the following requisites:

- [x] Prepare the directory structure
- [x] Implement the App component 
- [x] Implement the BooksList component
- [x] Implement the Book component
- [x] Implement the BooksForm component 
- [x] Define the Redux actions and reducers
- [x] Initialize the Redux store

### CODE REVIEW

**Comment by uryelah** [comment]
## Invalid Code Review Request 🛑 

Please use the same PR from milestone 1 to submit the changes of milestone 2.

According to the [project specifications:](https://microverse.pathwright.com/library/new-technical-curriculum/177956/path/step/104328875/)
Screenshot_2021-02-04 Microverse Program 2 0

Make sure the changes for M2 are visible in the first Pull Request, #1 , and submit that PR for the M2 review.

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,11 @@
...
-      </header>
-    </div>
-  );
-}
+const App = () => (
+  <div>
+    <BooksList />
+    <BooksForm />
+  </div>
+);
```
Comment (BrittanyBlake) [inline src/App.js]: - [ ] Your App component should be in the `src/components` directory
screenshot of requirements 👇🏾 
Screen Shot 2021-02-04 at 8 55 03 PM

File: src/components/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        </tr>
+
+        <BooksDetail />
+
+      </tbody>
+    </table>
+  </div>
+);
+
+export default BooksList;
```
Comment (BrittanyBlake) [inline src/components/BooksList.js]: - [ ] This should be placed in the `src/containers` directory.
- [ ] it should display books list in html table using `Book` component
- [ ] It should be connected to the redux store and fetch books from the state
 screenshot from pathwright 👇🏾 
Screen Shot 2021-02-04 at 9 01 04 PM

File: src/components/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <h3 htmlFor="title">Title</h3>
+    <input type="text" id="title" />
+    <select>
+      {CATEGORIES.map(category => <option key={category} value={category}>{category}</option>)}
+    </select>
+    <button type="button">Add Book</button>
+  </form>
+);
+
+export default BooksForm;
```
Comment (BrittanyBlake) [inline src/components/BooksForm.js]: - [ ] This should be placed in the `src/containers` directory

**Review by BrittanyBlake — CHANGES_REQUESTED** [review]
### Status: Changes Required ♻️ 
****
### Hi @ferbaco86! 👋 
You have done a wonderful job with your project so far! 🤩  It looks really good! 👏 👏 👏  There are just a few issues that need to be resolved before you can move on: ⬇️ 
****

- [ ] The `src/index.js` file should have example books in the initial state object
screenshot from pathwright 👇🏾 
Screen Shot 2021-02-04 at 8 52 44 PM

- [ ] It is important to follow the pattern on how to divide components into Containers and Presentational Components. Container components are the ones that are connected to the redux store, in this case, BookList and BookForm. Presentational components are the ones that do not communicate with redux.
screenshot from pathwright 👇🏾 
Screen Shot 2021-02-04 at 9 16 33 PM


****
### Please make the required changes and resubmit for a new code review once you are ready 🙏

Good Luck and Happy Coding 🍀 👩🏾‍💻

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Brittany Blake)._

**Review by PhillipUg — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on implementing all requirements and getting this milestone approved. :+1:

You can merge this branch into the development branch when you're ready and go on to your next milestone.
:bulb:  Do not forget to submit your project completion form after merging this branch.

All the best team,

For any assistance, you can contact me on slack as **Phillip Musiime** or
 <a href="mailto:phillipmusiime@gmail.com"> gmail</a> &nbsp; &nbsp; [Twitter Follow](https://twitter.com/intent/follow?screen_name=Phillip_Ug) &nbsp; &nbsp; [GitHub followers](https://github.com/PhillipUg)

**Pull request merged**